tests/integration/devfile/cmd_devfile_describe_test.go

```diff
@@ -107,4 +127,66 @@ var _ = Describe("odo devfile describe command tests", func() {
 		})
 	})
 
+	Context("devfile has missing metadata", func() {
+		// Reference: https://github.com/openshift/odo/issues/4815
+		var metadata devfilepkg.DevfileMetadata
+
+		// copyAndCreate copies required source code and devfile, and creates a component
+		var copyAndCreate = func(path string) {
+			// Using SpringBoot example here because it helps to distinguish between language and projectType.
+			// In terms of SpringBoot, spring in the projectType and java is the language
```

Comment:
Could you remove the hardcoding of Spring Boot example here? Try to make this more generic example by putting it into a helper file that makes sense? I'm asking this because we're copying source and devfile at a lot of places and I see this could be helpful in more generic cases than it would be if it were going to be a function limited to this `Context`.

Reply:
I would still like to make a note about why Spring Boot is used here, hence I've added a comment in the `Context("devfile has missing metadata", func(){})` block, I hope that is okay.

Reply:
Yeah. My request was more along the lines of making it generic so that it can be used in more places.